### PR TITLE
Merge in SVE splat intrinsic patch and add isel support for vaddi

### PR DESCRIPTION
**Pull request by pranav-prakash**
Should close off #12

### CODE REVIEW

**Comment by rkruppe** [comment]
The final result looks good to me, but I've noticed that some commits intermingle your changes with those by Graham. That loses authorship information and will complicate rebasing for us once Graham's patches are accepted upstream. Could you please reorganize the commits like this?

- import of D47768 (the commit message should mention D47768 and attribute it to Graham Hunter)
- import of D47775 (the commit message should mention D47775 and attribute it to Graham Hunter)
- your own changes

**Comment by pranav-prakash** [comment]
>Could you please reorganize the commits like this?
Alright done

File: lib/Target/RISCV/RISCVISelLowering.cpp [hunk 1]
```diff
@@ -525,9 +525,22 @@ SDValue RISCVTargetLowering::lowerINTRINSIC_WO_CHAIN(SDValue Op,
     return SDValue(); // Don't custom lower most intrinsics
   case Intrinsic::riscv_setvl:
     return lowerSETVL(Op, DAG);
+  case Intrinsic::experimental_vector_splatvector:
+    return lowerSPLAT_VECTOR(Op, DAG);
   }
 }
 
+SDValue RISCVTargetLowering::lowerSPLAT_VECTOR(SDValue Op, SelectionDAG &DAG) const {
+  SDLoc dl(Op);
```
Comment (rkruppe) [inline lib/Target/RISCV/RISCVISelLowering.cpp]: Sorry I missed this the first time around, but this should be DL.

Comment (pranav-prakash) [reply lib/Target/RISCV/RISCVISelLowering.cpp]: Fixed!

**Comment by pranav-prakash** [comment]
>What happened to the CodeGen test you had previously

Whoops seems I forgot to commit that after rebasing everything. Nice catch!

**Comment by rkruppe** [comment]
Thanks again!

**Pull request merged**